.github/workflows/publish.yml

```diff
@@ -58,41 +61,20 @@ jobs:
       test_command: pytest -p no:warnings --astropy-header -m "not hypothesis" -k "not test_data_out_of_range and not test_set_locale and not TestQuantityTyping" --pyargs astropy
       targets: |
         # Linux wheels
-
-        - cp310-manylinux_x86_64
-        - cp311-manylinux_x86_64
-        - cp312-manylinux_x86_64
+        - cp3*-manylinux_x86_64
```

Comment:
I think we want to explicitly control which Python versions we're building wheels for instead of relying on always being in sync with OpenAstronomy somehow? But maybe @astrofrog can confirm why these were separate in the first place.

Reply:
This is the only way I could find to express that I wanted cp313 wheels to build _conditionnally_, without having to change the OpenAstronomy action or ditching it entirely.
Minimal version is correctly parsed from pyproject.toml and max version only changes once a year, and only if we update cibuildwheel, so I'm not sure I see an advantage in expanding the list.

Reply:
I think I get it: by expanding the list, OpenAstronomy's publish workflow is able to create one job per wheel, so it's much more parallelized and you don't wait as long for all wheels to complete (also, presumably, they upload independently). This is a neat feature, but it doesn't play nice with cibuildwheel, in that if one selects cp313 wheels but `CIBW_PRERELEASE_PYTHON=false`, the job just fails
```
cibuildwheel: No build identifiers selected: BuildSelector(build_config='cp313-manylinux_x86_64', skip_config='', 
```
whereas if we just use cibuildwheel directly, this wheel is simply skipped.
I'll report to OpenAstronomy about that, but in case it turns out they cannot solve this, we can still work around it, either as I did (loosing parallelization) or by duplicating most of the configuration into branches. Any workaround can be reverted when 3.13.0rc1 is out.

Reply:
This also gives me motivation for splitting this PR in smaller pieces actually 🤔

Reply:
Here's the report https://github.com/OpenAstronomy/github-actions-workflows/issues/212
I'll now proceed splitting this PR.